xarray/backends/pseudonetcdf_.py

```diff
@@ -75,17 +75,17 @@ def get_variables(self):
                                  for k, v in self.ds.variables.items())
 
     def get_attrs(self):
-        return Frozen(dict([(k, getattr(self.ds, k))
-                            for k in self.ds.ncattrs()]))
+        return Frozen({k: getattr(self.ds, k) for k in self.ds.ncattrs()})
 
     def get_dimensions(self):
         return Frozen(self.ds.dimensions)
 
     def get_encoding(self):
         encoding = {}
-        encoding['unlimited_dims'] = set(
-            [k for k in self.ds.dimensions
-             if self.ds.dimensions[k].isunlimited()])
+        encoding['unlimited_dims'] = {
+            k for k in self.ds.dimensions
+            if self.ds.dimensions[k].isunlimited()
+        }
         return encoding
```

Comment:
This method could use the same, more concise, version as pynio from the next file.

Reply:
Done, thanks for spotting it